src/main/java/api/rest/v2/samples/SamplesControllerV2.java

```diff
@@ -0,0 +1,123 @@
+package api.rest.v2.samples;
+
+
+import application.ApplicationException;
+import application.SampleService;
+import domain.InvalidDomainException;
+import domain.notification.INotificationRepository;
+import domain.sample.SampleEventDatasource;
+import domain.sample.SampleEventStore;
+import domain.sample.SampleRepository;
+import domain.sample.Status;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.annotation.Body;
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.http.annotation.Error;
+import io.micronaut.http.annotation.Get;
+import io.micronaut.http.annotation.PathVariable;
+import io.micronaut.http.annotation.Put;
+import io.micronaut.security.annotation.Secured;
+import io.micronaut.security.rules.SecurityRule;
+import javax.inject.Inject;
+import life.qbic.auth.Authentication;
+
+@Requires(beans = Authentication.class)
+@Secured(SecurityRule.IS_AUTHENTICATED)
+@Controller("/v2/samples")
+public class SamplesControllerV2 {
+
+  SampleService sampleService;
+
+  @Inject
+  public SamplesControllerV2(SampleEventDatasource sampleEventDatasource, INotificationRepository notificationRepository) {
+    this.sampleService = new SampleService(
+        new SampleRepository(new SampleEventStore(sampleEventDatasource)), notificationRepository);
+  }
+
+  @Put(uri = "/{sampleCode}/status")
+  public HttpResponse<?> moveSampleToStatus(@PathVariable String sampleCode,
+      @Body StatusChangeRequest statusChangeRequest) {
+    String validFrom = statusChangeRequest.validFrom;
```

Comment:
NIT: "move" does imply a location to me, changeSampleStatus sounds more intuitive, imho

more critical: "validFrom" does not imply a time to me, especially since it is a String

Reply:
how would you call it? It is basically the instant from which the status is valid. I am open for suggestions here 😄

Reply:
maybe validFromTime or validSince?

Reply:
I like `validSince`:+1: